src/hooks/useActionCenterSubscriber/useTradeActionSubscriber.ts

```diff
@@ -0,0 +1,212 @@
+import type { QuoteId } from '@shapeshiftoss/types'
+import { fromBaseUnit } from '@shapeshiftoss/utils'
+import { useQueries } from '@tanstack/react-query'
+import { uuidv4 } from '@walletconnect/utils'
+import { useEffect, useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+
+import { useLocaleFormatter } from '../useLocaleFormatter/useLocaleFormatter'
+import { useWallet } from '../useWallet/useWallet'
+import { getTradeStatusHandler } from './checkStatusHandlers/getTradeStatusHandler'
+
+import { useCurrentHopIndex } from '@/components/MultiHopTrade/components/TradeConfirm/hooks/useCurrentHopIndex'
+import { actionCenterSlice } from '@/state/slices/actionSlice/actionSlice'
+import {
+  ActionCenterType,
+  ActionStatus,
+  isTradePayloadDiscriminator,
+} from '@/state/slices/actionSlice/types'
+import { selectPendingSwapActionsFilteredByWallet } from '@/state/slices/selectors'
+import { selectSwapById, selectSwapByQuoteId } from '@/state/slices/swapSlice/selectors'
+import { swapSlice } from '@/state/slices/swapSlice/swapSlice'
+import {
+  selectFirstHopSellAccountId,
+  selectInputBuyAsset,
+  selectInputSellAsset,
+} from '@/state/slices/tradeInputSlice/selectors'
+import {
+  selectActiveQuote,
+  selectConfirmedTradeExecution,
+} from '@/state/slices/tradeQuoteSlice/selectors'
+import { TradeExecutionState } from '@/state/slices/tradeQuoteSlice/types'
+import { store, useAppDispatch, useAppSelector } from '@/state/store'
+
+export const useTradeActionSubscriber = () => {
+  const dispatch = useAppDispatch()
+  const translate = useTranslate()
+
+  const {
+    number: { toCrypto },
+  } = useLocaleFormatter()
+
+  const tradeExecution = useAppSelector(selectConfirmedTradeExecution)
+  const tradeSellAsset = useAppSelector(selectInputSellAsset)
+  const tradeBuyAsset = useAppSelector(selectInputBuyAsset)
+  const tradeQuote = useAppSelector(selectActiveQuote)
+  const previousTradeExecutionState = usePrevious(tradeExecution?.state)
+  const previousTradeQuoteId = usePrevious(tradeQuote?.id)
+  const currentHopIndex = useCurrentHopIndex()
+  const previousTxHash = usePrevious(tradeExecution?.firstHop.swap.sellTxHash)
+  const sellAccountId = useAppSelector(selectFirstHopSellAccountId)
+  const previousQuoteOrRate = usePrevious(tradeQuote?.quoteOrRate)
+  const pendingSwapActions = useAppSelector(selectPendingSwapActionsFilteredByWallet)
+  const toast = useToast()
+  const swapByIds = useAppSelector(selectSwapById)
+  const {
+    state: { isConnected },
+  } = useWallet()
+
+  // Create swap and action after user confirmed the intent
+  useEffect(() => {
+    if (!tradeExecution) return
+
+    const firstStep = tradeQuote?.steps[0]
+    const lastStep = tradeQuote?.steps[tradeQuote.steps.length - 1]
+
+    if (!firstStep || !lastStep) return
+    if (!sellAccountId) return
+
+    if (tradeQuote.quoteOrRate === 'quote' && previousQuoteOrRate === 'rate') {
+      const swap: Swap = {
```

Comment:
Is this a good place for this? Doesn't seem like this is the responsability of a status poller

Reply:
Same as before:

Any suggestion as a replacement?

Upserting the swap directly in the swapper bits?